pandas/core/frame.py

```diff
@@ -2587,7 +2592,28 @@ def transpose(self, *args, **kwargs):
         dtype: object
         """
         nv.validate_transpose(args, dict())
-        return super().transpose(1, 0, **kwargs)
+        # construct the args
```

Comment:
don't you think this is better located in pandas/core/reshape ? (and called as a helper function here)

Reply:
Maybe a slight preference for keeping it here just for readability. The reshape part is essentially just a list comprehension.

```python
values = self.values
new_values = [arr_type._from_sequence(row, dtype=dtype) for row in values]
```

which I don't think warrants its own function. I don't see anything places in `/core/reshape.py` that could use this. I believe those are reshaping to / from 1-D things. This is 2D -> 2D.

But happy to move it if you want / if you see other places that could use parts of this.